eth_account/hdaccount/deterministic.py

```diff
@@ -0,0 +1,192 @@
+"""
+Heirarchical Deterministic Wallet generator (HDWallet)
+
+Partially implements the BIP-0032, BIP-0043, and BIP-0044 specifications:
+BIP-0032: https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki
+BIP-0043: https://github.com/bitcoin/bips/blob/master/bip-0043.mediawiki
+BIP-0044: https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki
+
+Skips serialization and public key derivation as unnecssary for this library's purposes.
+
+Notes
+-----
+
+* Integers are modulo the order of the curve (referred to as n).
+* Addition (+) of two coordinate pair is defined as application of the EC group operation.
+* Concatenation (||) is the operation of appending one byte sequence onto another.
+
+
+Definitions
+-----------
+
+* point(p): returns the coordinate pair resulting from EC point multiplication
+  (repeated application of the EC group operation) of the secp256k1 base point
+  with the integer p.
+* ser_32(i): serialize a 32-bit unsigned integer i as a 4-byte sequence,
+  most significant byte first.
+* ser_256(p): serializes the integer p as a 32-byte sequence, most significant byte first.
+* ser_P(P): serializes the coordinate pair P = (x,y) as a byte sequence using SEC1's compressed
+  form: (0x02 or 0x03) || ser_256(x), where the header byte depends on the parity of the
+  omitted y coordinate.
+* parse_256(p): interprets a 32-byte sequence as a 256-bit number, most significant byte first.
+
+"""
+# Additional notes:
+# - This algorithm only implements private parent key → private child key CKD function,
+#   as it is unnecessary to the HD key derivation functions used in this library to implement
+#   the other functions (as Ethereum uses an Account-based system)
```

Comment:
This is 100% my own ignorance, but why does Ethereum using an Account-based system impact why we don't need the other CKD functions? It seems like we would need at least public -> public?

Reply:
Your guess is as good as mine for why people decided to take this path. Probably to be interoperable with Hardware wallets, which are built using HD Path derivation (for multiple coins)

Reply:
Actually, this comment is wrong. Account-based system doesn't really matter, I think I was trying to explain that the use of change addresses and continuous key generation (as a UTXO model would use to prevent key reuse) isn't a necessary factor here. But that seems like unnecessary information.